src/lib.rs

```diff
@@ -168,7 +168,7 @@ pub fn firefox_version(binary: &Path) -> Result<AppVersion, Error>  {
     let mut version = AppVersion::new();
     let mut updated = false;
 
-    if let Some(dir) = ini_path(binary) {
+    if let Some(dir) = ini_path(&binary.canonicalize().unwrap()) {
```

Comment:
It returns a `Result<PathBuf, std::io::Error>`. `Result<T>` in that crate is a typedef for `Result<T, std::io::Error>`, which is a little confusing until you are used to it.